sdcm/nemesis.py

```diff
@@ -5006,6 +5008,113 @@ def disrupt_disable_binary_gossip_execute_major_compaction(self):
             self.target_node.restart_scylla_server()
             raise
 
+    def disrupt_refusing_connection_from_banned_node(self, use_iptables=False):
+        """Banned node could not connect with rest nodes in cluster
+
+        If node was removed from cluster for any reason, even if removed node
+        become alive and try to communicate with rest node in cluster, all connections
+        from it should be refused by other nodes in cluster
+        1. on target node block any scylladb process
+        1.1 Pause process
+        1.2 Block with iptables port 9100/10000
+        2. Wait and remove target node from cluster
+        3. start scylla on target node
+        4. Create exclusive connection to target node
+        5. Execute cql command on target node and validate that no operation
+        from target node passed to cluster
+        """
+        if not self.target_node.raft.consistent_topology_changes_enabled:
+            raise UnsupportedNemesis("Raft feature: consistent-topology-changes is not enabled")
+
+        def prepare_test_keyspace(node):
+            """ Create new table for verification missing data
+
+            Create new keyspace and table which allow to
+            verify that no operations come from banned node
+            """
+            with self.cluster.cql_connection_patient(node) as session:
+                session.execute(
+                    f"CREATE KEYSPACE IF NOT EXISTS banned_keyspace \
+                        WITH replication = {{'class': 'NetworkTopologyStrategy', \
+                            'replication_factor' : {self.tester.params.get('n_db_nodes')} }};")
+                session.execute("CREATE TABLE IF NOT EXISTS banned_keyspace.table1 (id BIGINT PRIMARY KEY, name text);")
+
+        simulate_node_unavailability = node_operations.simulate_node_unavailability(use_iptables)
```

Comment:
`node_unavailable` = `'blocked_scylla_ports' if use_iptables else 'scylla_stopped_by_sigstop'`

I don't see value in hiding this in function.

Reply:
Removed extra function